src/tools/tests/allocator-recipe-resolver-test.ts

```diff
@@ -1149,6 +1157,167 @@ recipe ReadingRecipeAD
       ${recipe}
     `, 'Thing {a: (Text, Number), c: List<(Text, Number)>, b: [(Text, Number)], d: (Text, Number)}');
   });
+
+  it.only('restricts writer fields with phantom readers', async () => {
+    const verifyPhantomReader = async (
+      schemas: string, policy: string, writes: string,  expected: string) => {
+      // Note that this should be intended as the other multiline strings in this doc.
+      const recipe = `
+      ${schemas}
+
+      particle Writer
+        data: writes ${writes}
+
+      @arcId('writeArcId')
+      recipe WritingRecipe
+        thing: create 'my-handle-id' @persistent @ttl('2d')
+        Writer
+          data: writes thing
+      `;
+      const policies = `
+      ${schemas}
+
+      policy Policy {
+         @maxAge(age: '2d')
+         @allowedRetention(medium: 'Disk', encryption: false)
+         ${policy}
+      }
+      `;
+      await verifyWritingRecipeWithPolicy(recipe, policies, expected);
+    };
+
+    const thingSchema = `
+      schema Thing
+        a: Text
+        b: Text
+        c: Text
+        d: Text
+        e: Text
+    `;
+    await verifyPhantomReader(
+      thingSchema,
+      'from Thing access {a, b, d}',
+      'Thing',
```

Comment:
The best practice is to use inline schemas in particles. The benefit here would also be that you show what's written, and what's expected next to each other.

E.g.

```
    await verifyPhantomReader(
      thingSchema,
      'from Thing access {a, b, d}',
      'Thing {a, b, c, d, e}',
      'Thing {a: Text, b: Text, d: Text}'
    );
```

I would maybe refactor the method for readability like this:
```
    await verifyPhantomReader({
      schema: thingSchema,
      policy: 'from Thing access {a, b, d}',
      written: 'Thing {a, b, c, d, e}',
      expected: 'Thing {a: Text, b: Text, d: Text}'
    });
```

I've seen that Josh has been working to enable { } at the schema level, so maybe you could even inline reasonably:

```
    await verifyPhantomReader({
      schema: 'schema Thing {a: Text, b: Text, c:Text}',
      policy: 'from Thing access {a, c}',
      written: 'Thing {a, b, c}',
      expected: 'Thing {a: Text, c: Text}'
    });
```

Reply:
Thanks for the suggestion, Piotr. I have refactored the tests for readability.